### DISCUSSION TITLE
Entity 단방향 매핑일때 한번에 조회하는 방법?

### DISCUSSION THREAD

**Discussion by Minoolian** [ProjectCoco/wayfarer-backend#9] [Dive Deep]
Entity간 연관관계에서 단방향 매핑으로 연결했을 때, `FK를 가진쪽(연관관계의 주인)이 아닌 Entity`에서는 어떻게 조회를 하면 효율적일까요?

단순히 생각나는 방법은 Entity의 ***Repository를 통해*** 찾는 방법 또는 ***양방향 연관관계***를 맺어 조회하는 방법밖에는 생각이나지 않네요. 물론 복잡한 의존성을 피하기 위해 가능한 양방향 매핑은 **지양**하기로한 우리의 생각을 고려했습니다.

**Comment by 5witchkr** [comment]
ddd에서 애그리거트에대한 부분을 좀 더 찾아봐야할것같아요
애그리거트루트는 애그리거트 요소들을 벨류형태로 가지고있고,
한 애그리거트는 한 레포지토리가 관리하는걸로 알고있어요.

**Reply by Minoolian** [reply]
https://sgc109.github.io/2020/08/09/ddd-basic/
이 글에서 DDD에서 사용하는 용어들을 잘 정리해놓았는데. 스위치님이 이에 대해 잘 설명해 주신것 같아요.
저도 Entity, Value에 관해서 더 알아보고 적용하겠습니다.
